src/go/rpk/pkg/tuners/gcp_disk_write_cache_tuner_test.go

```diff
@@ -0,0 +1,200 @@
+	// when
+	tuner.Tune()
+	// then
+	setValue, _ := afero.ReadFile(fs, devicePath+"/queue/write_cache")
+	require.Equal(t, "write through", string(setValue))
+}
+
+func TestDeviceWriteCacheTuner_IsSupported_Should_return_true(t *testing.T) {
+	// given
+	v := &vendorMock{
+		init: func() (vendor.InitializedVendor, error) {
+			return &currentVendor{
+				name: "gcp",
+			}, nil
+		},
+	}
+	deviceFeatures := &deviceFeaturesMock{
+		getWriteCacheFeatureFile: func(string) (string, error) {
+			return devicePath + "/queue/write_cache", nil
+		},
+		getWriteCache: func(string) (string, error) {
+			return "write back", nil
+		},
+	}
+	fs := afero.NewMemMapFs()
+	fs.MkdirAll(devicePath+"/queue", 0644)
+	tuner := NewDeviceGcpWriteCacheTuner(fs, "fake", deviceFeatures, v, executors.NewDirectExecutor())
+	// when
+	supported, _ := tuner.CheckIfSupported()
+	// then
+	require.True(t, supported)
+}
+
+func TestDeviceWriteCacheTuner_IsSupported_should_return_false_on_aws(
+	t *testing.T,
+) {
+	// given
+	v := &vendorMock{
+		init: func() (vendor.InitializedVendor, error) {
+			return &currentVendor{
+				name: "aws",
+			}, nil
+		},
+	}
+	deviceFeatures := &deviceFeaturesMock{
+		getWriteCacheFeatureFile: func(string) (string, error) {
+			return devicePath + "/queue/write_cache", nil
+		},
+		getWriteCache: func(string) (string, error) {
+			return "write back", nil
+		},
+	}
+	fs := afero.NewMemMapFs()
+	fs.MkdirAll(devicePath+"/queue", 0644)
+	tuner := NewDeviceGcpWriteCacheTuner(fs, "fake", deviceFeatures, v, executors.NewDirectExecutor())
+	// when
+	supported, _ := tuner.CheckIfSupported()
+	// then
+	require.False(t, supported)
+}
+
+func TestGCPCacheTunerSupported(t *testing.T) {
+	type args struct {
+		fs             afero.Fs
+		device         string
+		deviceFeatures disk.DeviceFeatures
+		vendor         vendor.Vendor
+		executor       executors.Executor
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "should be supported on GCP",
+			args: args{
+				fs:     afero.NewMemMapFs(),
+				device: "fake",
+				vendor: &vendorMock{
+					init: func() (vendor.InitializedVendor, error) {
+						return &currentVendor{
+							name: "gcp",
+						}, nil
+					},
+				},
+				deviceFeatures: &deviceFeaturesMock{},
+				executor:       executors.NewDirectExecutor(),
+			},
+			want: true,
+		},
+		{
+			name: "should not be supported on AWS",
+			args: args{
+				fs:     afero.NewMemMapFs(),
+				device: "fake",
+				vendor: &vendorMock{
+					init: func() (vendor.InitializedVendor, error) {
+						return &currentVendor{
+							name: "aws",
+						}, nil
+					},
+				},
+				deviceFeatures: &deviceFeaturesMock{},
+				executor:       executors.NewDirectExecutor(),
+			},
+			want: false,
+		},
+		{
+			name: "should not be supported on not cloud deployments",
+			args: args{
+				fs:     afero.NewMemMapFs(),
+				device: "fake",
```

Comment:
If the values for `fs`, `device`, `deviceFeatures` & `executor` are always the same, there's no need to add them to the struct.